runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/internal/stream/KafkaClientProduceFactory.java

```diff
@@ -1812,7 +1824,14 @@ private void doEncodeRecordFin(
 
                 if (encodeableRecordBytesDeferred > 0)
                 {
-                    doNetworkData(traceId, budgetId, EMPTY_BUFFER, 0, 0);
+                    if (flushableRequestBytes == 0)
+                    {
+                        doEncodeRequestIfNecessary(traceId, budgetId);
+                    }
```

Comment:
This logic is proactively encoding the produce request when the incoming message is fragmented.

It seems to be defeating message batching because it doesn't give a chance for more message fragments and messages to arrive before sending the batched request. If no more message fragments or messages arrive, then the signal should encode the produce request anyway.

I think the original mental model for this assumed messages would only be fragmented due to flow control forcing it when the encode slot is full, but we are not explicitly verifying that so when small messages are fragmented we are unnecessarily (and incorrectly) proactively sending the produce request.

Please verify that batching is still working as expected and adjust this logic as needed.